### PR TITLE
Add docs to create and update partner

### PR DESCRIPTION
**Pull request by ahsan-ul-haq**
ECOM-2235
@zubair-arbi @aamir-khan @awais786 @tasawernawaz 
- Add basic docs for model `Partner` in the app `partner`.
- Add screen shots of model's django admin,

Code changes (PR merged) https://github.com/edx/ecommerce/pull/325 https://github.com/edx/ecommerce/pull/330
Preview: https://github.com/edx/ecommerce/blob/ahsan/ECOM-2235-Update-Partner-Model-docs/docs/partner_config.rst

### CODE REVIEW

File: docs/partner_config.rst [hunk 1]
```diff
@@ -26,3 +26,28 @@ The following image shows the ``SiteConfiguration`` model in the Django administ
...
 
     Please make sure that there is only one partner per site.
+
+
+======================
+Partner Configuration
+======================
+
+This is additional configuration in the ``Partner`` model to allow to add **short code** that would be a unique identifier for ``Partner``.
+This additional configuration is necessary to create new partner which will be further use in the ``SiteConfiguration`` model for specific theming.
```
Comment (zubair-arbi) [inline docs/partner_config.rst]: This is an additional configuration in the Partner model to allow adding a short code that would be a unique identifier for that Partner. This configuration is necessary to create a new partner which will be further used in different places, e.g. in the SiteConfiguration model for specific theming.

File: docs/partner_config.rst [hunk 2]
```diff
@@ -26,3 +26,28 @@ The following image shows the ``SiteConfiguration`` model in the Django administ
...
+---------------------------------------------
+
+To add and update partner's configurations use the ``Partner`` model in the `Django administration panel <http://localhost:8002/admin/partner/partner/add/>`_.
+This panel is located at http://0.0.0.0:8002/admin/partner/partner/.
+
+The following images shows the ``Partner`` model in the Django administration panel for a configured partner.
+
+.. image:: images/partner_configuration.png
+    :width: 600px
+    :alt: Populated site configuration model
```
Comment (zubair-arbi) [inline docs/partner_config.rst:48]: nit: change the `alt` content

**Comment by zubair-arbi** [comment]
lgtm :+1:

**Comment by ahsan-ul-haq** [comment]
@srpearce @clintonb could you please review this PR

File: docs/partner_config.rst [hunk 3]
```diff
@@ -26,3 +26,28 @@ The following image shows the ``SiteConfiguration`` model in the Django administ
...
 
     Please make sure that there is only one partner per site.
+
+
+======================
+Partner Configuration
+======================
+
+This is an additional configuration in the ``Partner`` model to allow adding a ``short code`` that would be a **unique identifier** for that ``Partner``.
+This configuration is necessary to create a new partner which will be further used in different places, e.g. in the ``SiteConfiguration`` model for specific theming.
```
Comment (srpearce) [inline docs/partner_config.rst]: This additional configuration in the `Partner` model allows users to specify a short code that serves as a unique identifier for that partner.

This required configuration creates a new **Partner** object that will be used throughout the platform, such as in the `SiteConfiguration` model for specific theming.

File: docs/partner_config.rst [hunk 4]
```diff
@@ -26,3 +26,28 @@ The following image shows the ``SiteConfiguration`` model in the Django administ
...
+
+
+---------------------------------------------
+Partner Configuration Model | Django Admin
+---------------------------------------------
+
+To add and update partner's configurations use the ``Partner`` model in the `Django administration panel <http://localhost:8002/admin/partner/partner/add/>`_.
```
Comment (srpearce) [inline docs/partner_config.rst]: update a partner's configurations

File: docs/partner_config.rst [hunk 5]
```diff
@@ -26,3 +26,28 @@ The following image shows the ``SiteConfiguration`` model in the Django administ
...
+This panel is located at http://0.0.0.0:8002/admin/partner/partner/.
+
+The following image shows the ``Partner`` model in the Django administration panel for a configured partner.
+
+.. image:: images/partner_configuration.png
+    :width: 600px
+    :alt: Populated partner model
+
+.. note::  There is a **unique** and **not null** constraint on the **short code** field for the ``Partner`` model.
+    This means that **short code** field is **required** and should be **unique** for each partner.
```
Comment (srpearce) [inline docs/partner_config.rst]: There is a **unique** and **not null** constraint on the **Short code** field for the `Partner` model. This means that the **Short code** field is required and must be unique for each partner.

**Comment by srpearce** [comment]
Doc review done.

**Comment by ahsan-ul-haq** [comment]
@srpearce I have made the changes please review

File: docs/partner_config.rst [hunk 6]
```diff
@@ -26,3 +26,28 @@ The following image shows the ``SiteConfiguration`` model in the Django administ
...
+
+This additional configuration in the ``Partner`` model allows users to specify a short code that serves as a unique identifier for that partner.
+This required configuration creates a new **Partner** object that will be used throughout the platform, such as in the ``SiteConfiguration`` model for specific theming.
+
+
+---------------------------------------------
+Partner Configuration Model | Django Admin
+---------------------------------------------
+
+To add and update a partner's configurations use the ``Partner`` model in the `Django administration panel <http://localhost:8002/admin/partner/partner/add/>`_.
```
Comment (srpearce) [inline docs/partner_config.rst]: Add a comma after configurations

**Comment by srpearce** [comment]
Hi @ahsan-ul-haq - Just one very small change, and then :+1: from me.

File: docs/partner_config.rst [hunk 7]
```diff
@@ -26,3 +26,28 @@ The following image shows the ``SiteConfiguration`` model in the Django administ
...
+This additional configuration in the ``Partner`` model allows users to specify a short code that serves as a unique identifier for that partner.
+This required configuration creates a new **Partner** object that will be used throughout the platform, such as in the ``SiteConfiguration`` model for specific theming.
+
+
+---------------------------------------------
+Partner Configuration Model | Django Admin
+---------------------------------------------
+
+To add and update a partner's configurations, use the ``Partner`` model in the `Django administration panel <http://localhost:8002/admin/partner/partner/add/>`_.
+This panel is located at http://0.0.0.0:8002/admin/partner/partner/.
```
Comment (clintonb) [inline docs/partner_config.rst]: http://localhost:8002

File: docs/partner_config.rst [hunk 8]
```diff
@@ -26,3 +26,28 @@ The following image shows the ``SiteConfiguration`` model in the Django administ
...
+
+This additional configuration in the ``Partner`` model allows users to specify a short code that serves as a unique identifier for that partner.
+This required configuration creates a new **Partner** object that will be used throughout the platform, such as in the ``SiteConfiguration`` model for specific theming.
+
+
+---------------------------------------------
+Partner Configuration Model | Django Admin
+---------------------------------------------
+
+To add and update a partner's configurations, use the ``Partner`` model in the `Django administration panel <http://localhost:8002/admin/partner/partner/add/>`_.
```
Comment (clintonb) [inline docs/partner_config.rst]: Partners can be created or modified in the Django administration site located at http://localhost:8002/admin/partner/partner/.

File: docs/partner_config.rst [hunk 9]
```diff
@@ -26,3 +26,28 @@ The following image shows the ``SiteConfiguration`` model in the Django administ
...
+This panel is located at http://0.0.0.0:8002/admin/partner/partner/.
+
+The following image shows the ``Partner`` model in the Django administration panel for a configured partner.
+
+.. image:: images/partner_configuration.png
+    :width: 600px
+    :alt: Populated partner model
+
+.. note::  There is a **unique** and **not null** constraint on the **Short code** field for the ``Partner`` model. This means that the **Short code** field is required and must be unique for each partner.
```
Comment (clintonb) [inline docs/partner_config.rst]: The short code field must be unique across all partners. This value is presently used as the order number prefix (e.g. "EDX" in EDX-1000).

File: docs/partner_config.rst [hunk 10]
```diff
@@ -26,3 +26,28 @@ The following image shows the ``SiteConfiguration`` model in the Django administ
     This means that there can be only one entry per site per partner.
 
     Please make sure that there is only one partner per site.
+
+
+======================
```
Comment (clintonb) [inline docs/partner_config.rst:26]: Move this section above SiteConfiguration.

**Comment by clintonb** [comment]
:+1: (once changes addressed)

**Pull request merged**